jormungandr-watchdog/src/service/control.rs

```diff
@@ -0,0 +1,82 @@
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tokio::{
+    stream::Stream,
+    sync::watch::{self, Receiver, Sender},
+};
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+pub enum Control {
+    Shutdown,
+    Kill,
+}
+
+pub struct Controller {
+    sender: Sender<Control>,
+    receiver: Receiver<Control>,
+}
+
+pub struct ControlReader {
+    receiver: Receiver<Control>,
```

Comment:
`watch::Receiver` is only guaranteed to receive the last item. Is this intended for `Control` command channel to be lossy in this way: only the last command matters?

Reply:
So far this is my finding. Trying to do something else was more complex as it needed to store more state in the service. However this is still open for comment/discussion.

Reply:
Also, there is only 2 control commands: `Shutdown` and `Kill`. Usually if you miss the `Shutdown` and it is now time to `Kill` missing the first one is not a problem but actually a welcome simplification of the process of handling the control commands.